Sources/Subscription/API/SubscriptionEndpointService.swift

```diff
@@ -63,118 +71,189 @@ public protocol SubscriptionEndpointService {
+            updateCache(with: subscription)
+        } else {
+            Logger.subscriptionEndpointService.debug("No subscription update required")
+        }
+        return subscription
+    }
+
+    func updateCache(with subscription: PrivacyProSubscription) {
+        cacheSerialQueue.sync {
             subscriptionCache.set(subscription)
+            Logger.subscriptionEndpointService.debug("Notifying subscription changed")
             NotificationCenter.default.post(name: .subscriptionDidChange, object: self, userInfo: [UserDefaultsCacheKey.subscription: subscription])
         }
     }
 
-    public func getSubscription(accessToken: String, cachePolicy: APICachePolicy = .returnCacheDataElseLoad) async -> Result<Subscription, SubscriptionServiceError> {
+    public func ingestSubscription(_ subscription: PrivacyProSubscription) async throws {
+        try await storeAndAddFeaturesIfNeededTo(subscription: subscription)
+    }
 
+    public func getSubscription(accessToken: String, cachePolicy: SubscriptionCachePolicy = .returnCacheDataElseLoad) async throws -> PrivacyProSubscription {
         switch cachePolicy {
         case .reloadIgnoringLocalCacheData:
-            return await getRemoteSubscription(accessToken: accessToken)
+            return try await getRemoteSubscription(accessToken: accessToken)
 
         case .returnCacheDataElseLoad:
-            if let cachedSubscription = subscriptionCache.get() {
-                return .success(cachedSubscription)
+            if let cachedSubscription = getCachedSubscription() {
+                return cachedSubscription
             } else {
-                return await getRemoteSubscription(accessToken: accessToken)
+                return try await getRemoteSubscription(accessToken: accessToken)
             }
 
         case .returnCacheDataDontLoad:
-            if let cachedSubscription = subscriptionCache.get() {
-                return .success(cachedSubscription)
+            if let cachedSubscription = getCachedSubscription() {
+                return cachedSubscription
             } else {
-                return .failure(.noCachedData)
+                throw SubscriptionEndpointServiceError.noData
             }
         }
     }
 
-    public func signOut() {
-        subscriptionCache.reset()
+    private func getCachedSubscription() -> PrivacyProSubscription? {
+        var result: PrivacyProSubscription?
+        cacheSerialQueue.sync {
+            result = subscriptionCache.get()
+        }
+        return result
     }
 
-    // MARK: -
-
-    public func getProducts() async -> Result<[GetProductsItem], APIServiceError> {
-        await apiService.executeAPICall(method: "GET", endpoint: "products", headers: nil, body: nil)
+    public func clearSubscription() {
+        cacheSerialQueue.sync {
+            subscriptionCache.reset()
+        }
+//        NotificationCenter.default.post(name: .subscriptionDidChange, object: self, userInfo: nil)
```

Comment:
Please double check if this is needed.

Reply:
Not needed for now, anything related to the subscription cancellation is handled by the signOut that sends the proper notification